pkg/auth/internal/tokenstore/tokenstore.go

```diff
@@ -53,3 +97,25 @@ func (s *Store) DeleteToken(issuer string, clientID string) error {
 
 	return os.Remove(path)
 }
+
+func (s *Store) readData(issuer, clientID string) (storeData, error) {
+	path := s.path(issuer, clientID)
+	var data storeData
+	err := readJSONFile(path, &data)
+	if errors.Is(err, os.ErrNotExist) {
+		return storeData{Version: storeDataVersion}, nil
+	} else if err != nil {
+		return storeData{Version: storeDataVersion}, err
+	}
+	return data, nil
+}
+
+func (sd *storeData) addToken(tok *session.Token) {
+	tokens := []*session.Token{tok}
+	for _, t := range sd.Tokens {
+		if t.IDClaims().Subject != tok.IDClaims().Subject {
+			tokens = append(tokens, t)
+		}
+	}
+	sd.Tokens = tokens
```

Comment:
The order seems to matter and have product implications in terms of what is returned from ReadToken. Given that:

Can we distinguish login-writes from refresh-writes? 

I can understand wanting to move login-write-tokens to the front for priority. However, its unclear to me that refresh-write-tokens should move to the front, or whether they should be kept in their current position

Reply:
will do, this is a bug in my implementation. Refreshing a token should not change the order, only logins should.